pac4j-core/src/main/java/org/pac4j/core/util/CommonHelper.java

```diff
@@ -236,102 +236,156 @@ public static String toString(final Class<?> clazz, final Object... args) {
+        URLConnection con = null;
+        try {
+            URL url = new URL(name);
+
+            con = url.openConnection();
+
+            return con.getInputStream();
+        } catch (IOException ex) {
+            // Close the HTTP connection (if applicable).
+            if (con instanceof HttpURLConnection) {
+                ((HttpURLConnection) con).disconnect();
+            }
+            throw new TechnicalException(ex);
+        }
+    }
+
+    public static Resource getResource(final String filePath) {
+        return new WritableResource() {
 
             @Override
             public File getFile() {
-                return new File(filePath);
+                final String filename = getFilename();
+                if (filename != null) {
+                    return new File(filename);
+                }
+                return null;
             }
 
             @Override
-			public InputStream getInputStream() throws IOException {
-				return getInputStreamFromName(filePath);
-			}
-
-			@Override
-			public String getFilename() {
-				return filePath;
-			}
-
-			@Override
-			public boolean exists() {
-				return getFile().exists();
-			}
-			
-			@Override
-			public OutputStream getOutputStream() throws IOException {
-				return new FileOutputStream(filePath);
-			}
-
-		};
-	}
+            public InputStream getInputStream() throws IOException {
+                return getInputStreamFromName(filePath);
+            }
+
+            @Override
+            public String getFilename() {
+                String filename = null;
+                final String prefix = extractPrefix(filePath);
+                if (prefix == null) {
+                    filename = filePath;
+                } else if (prefix.equals(RESOURCE_PREFIX) || prefix.equals(CLASSPATH_PREFIX)) {
+                    final String path = filePath.substring(prefix.length() + 1);
+                    final URL url;
+                    if (prefix.equals(RESOURCE_PREFIX)) {
+                        url = CommonHelper.class.getResource(startWithSlash(path));
+                    } else {
+                        url = Thread.currentThread().getContextClassLoader().getResource(path);
+                    }
+                    if (url == null) {
+                        throw new TechnicalException("Do not use the resource: or classpath: prefix for non-existing files. Use a direct path (relative or absolute, no prefix)");
+                    }
+                    // remove the file: prefix
+                    filename = url.toString().substring(5);
+                }
+                logger.debug("filepath: {} -> filename: {}", filePath, filename);
+                return filename;
+            }
+
+            @Override
+            public boolean exists() {
+                final File f = getFile();
+                if (f != null) {
+                    return f.exists();
+                }
+                return true;
```

Comment:
Shouldn't this be false? This is odd. If we don't have a file, we assume it exists? Are you assuming this a stream?